File: src/models/Task.ts
import uuid from 'react-native-uuid';

export enum TaskStatus {
  TODO = 'TODO',
  IN_PROGRESS = 'IN_PROGRESS',
  COMPLETED = 'COMPLETED'
}

export enum TaskPriority {
  LOW = 'LOW',
  MEDIUM = 'MEDIUM',
  HIGH = 'HIGH'
}

export interface Task {
  id: string;
  projectId: string;
  title: string;
  description: string;
  status: TaskStatus;
  priority: TaskPriority;
  dueDate: Date | null;
  createdAt: Date;
}

export const createTask = (
  projectId: string,
  title: string,
  description: string,
  status: TaskStatus = TaskStatus.TODO,
  priority: TaskPriority = TaskPriority.MEDIUM,
  dueDate: Date | null = null
): Task => ({
  id: uuid.v4().toString(),
  projectId,
  title,
  description,
  status,
  priority,
  dueDate,
  createdAt: new Date()
}); 

File: app/src/main/java/com/example/pms/data/models/Task.kt
package com.example.pms.data.models

import java.util.Date
import java.util.UUID

data class Task(
    val id: String = UUID.randomUUID().toString(),
    val projectId: String,
    val title: String,
    val description: String,
    val status: TaskStatus,
    val priority: TaskPriority,
    val dueDate: Date?,
    val createdAt: Date = Date()
)

enum class TaskStatus {
    TODO,
    IN_PROGRESS,
    COMPLETED
}

enum class TaskPriority {
    LOW,
    MEDIUM,
    HIGH
}
